doc/visual-programming/source/widgets/data/rank.md

```diff
@@ -11,18 +11,20 @@ Ranking of attributes in classification or regression datasets.
 **Outputs**
 
 - Reduced Data: dataset with selected attributes
+- Scores: data table with feature scores
+- Features: list of attributes
 
-The **Rank** widget considers class-labeled datasets (classification or regression) and scores the attributes according to their correlation with the class. Rank accepts also models for scoring, such as linear regression, logistic regression, random forest, SGD, etc.
+The **Rank** widget considers class-labeled datasets (classification or regression) and unlabelled data (unsupervised). It scores the attributes according to their correlation with the class or in the case of unsupervised data, by variance explained by each principal component. Rank accepts also models for scoring, such as linear regression, logistic regression, random forest, SGD, etc.
```

Comment:
> by variance explained by each principal component

Please someone confirm this.

Reply:
> by variance explained by each principal component

No, because it doesn't rank principal components. It ranks by correlation with principal components. (Which is not very meaningful in terms of scoring, IMHO.)

What about the following formulation?

*The **Rank** widget scores variables according to their correlation with discrete or numeric target variable, based on applicable internal scorers (like information gain, chi-square and linear regression) and any connected external models that supports scoring, such as linear regression, logistic regression, random forest, SGD, etc. The widget can handle unsupervised data, but only by external scorers, such as PCA, which ranks the variables according to their correlation with individual principal components.*

Reply:
Ok, so it is an average score of variable's correlation with N principal components? Or just the first? What is N? 2? 😱

Reply:
`With individual principal components` should perhaps be `with chosen individual principal component`.

The number of components is chosen in the PCA widget, but describing this is out of this context, I think. Perhaps we could even cut the sentence at "PCA".

(((The average correlation with the first N components would be interesting - in some form - I guess. It would be a mearuse of outlierness (well, inlierness) of a feature. If you imagine a feature space as a flattish spheroid in n-dimensional space, this correlation could measure whether the feature is aligned with others or does it stick out of the "pancake".)))